tests/test_live_server.py

```diff
@@ -58,6 +58,53 @@ def test_a(live_server):
         result.stdout.fnmatch_lines(['*PASSED*'])
         assert result.ret == 0
 
+    def test_clean_stop_live_server(self, appdir):
+        pytest_cov = pytest.importorskip("pytest_cov")
+        appdir.create_test_module('''
+            import pytest
+            try:
+                from urllib2 import urlopen
+            except ImportError:
+                from urllib.request import urlopen
+
+            from flask import url_for
+
+            def test_a(live_server):
+                @live_server.app.route('/')
+                def index():
+                    return 'got it', 200
+
+                live_server.start()
+
+                res = urlopen(url_for('index', _external=True))
+                assert res.code == 200
+                assert b'got it' in res.read()
+        ''')
+        result_with = appdir.runpytest('-v',
```

Comment:
I think depending on coverage here is a little too brittle. 

I see a few options:
* monkeypatching `LiveServer.stop` to ensure it was called at the end of the session instead. 
* Use a "verbose" mode which prints to the output that it has been cleanly shutdown, then check the output by passing `-s` to pytest.